edgelet/iotedge-proxy/src/proxy/client.rs

```diff
@@ -94,6 +87,31 @@ where
     }
 }
 
+fn build_uri(destination_url: Url, requested_uri: &Uri) -> Result<Uri, Error> {
```

Comment:
Maybe rename `destination_url` to `base_url` (since from the POV of the caller it's just the base URL), and rename the `url` on line 104 to `destination_url` (since inside the function it becomes the destination URL) ?